ansible_base/lib/testing/util.py

```diff
@@ -36,3 +43,29 @@ def delete_authenticator(authenticator):
             pass
         au.delete()
     authenticator.delete()
+
+
+class StaticResourceAPIClient(ResourceAPIClient):
+    """A testing API client that reads response router attribute or static files."""
+
+    router = {}
+    # Route is used to force a certain status,response for a route
+    # It has to be a mutable default but the fixture instantiates one for
+    # each test.
+
+    def _make_request(self, method, path, data=None, params=None, stream=False):
+        response = Response()
+        response.encoding = "utf-8"
+
+        if path in self.router:
+            response.status_code = self.router[path]["status_code"]
+            response._content = self.router[path]["content"]
+            return response
+
+        content_file_path = Path(self.base_url) / path / "response"
+        response._content = content_file_path.read_bytes()
```

Comment:
Could you explain what's happening here? Its parent class does `resp = requests.request(**kwargs)` which is intuitively logically, this is a client, it makes a request to a live(?) server. But this just reads from... a file path? It reads like a URL turned into a path.

Reply:
It reads mocking/fixtures from static files for testing, as we cannot have a RESOURCE_SERVER running on CI

Reply:
The path `https://.../api/resource-types/shared.user/manifest/` is translated to `file://.../api/resource-types/shared.user/manifest/response|status`